checker/backward_compatibility_component_error.go

```diff
@@ -0,0 +1,107 @@
+package checker
+
+import (
+	"fmt"
+
+	"github.com/TwiN/go-color"
+	"github.com/tufin/oasdiff/checker/localizations"
+)
+
+type BackwardCompatibilityComponentError struct {
```

Comment:
```suggestion
type ComponentChange struct {
```

since we're here, do we want to rename this given it's not necessarily an error but rather a change of level info, error, warning? 

Other options
- OpenAPISpecChange
- SpecChange

Reply:
Good point